internal/handlers/handler_register_totp.go

```diff
@@ -37,19 +38,29 @@ var SecondFactorTOTPIdentityStart = middlewares.IdentityVerificationStart(middle
 })
 
 func secondFactorTOTPIdentityFinish(ctx *middlewares.AutheliaCtx, username string) {
+	algorithm := otp.AlgorithmSHA1
+
+	switch ctx.Configuration.TOTP.Algorithm {
+	case "sha256":
+		algorithm = otp.AlgorithmSHA256
+	case "sha512": // nolint: goconst
```

Comment:
```suggestion
	case "sha512":
```

This shouldn't be necessary anymore because it's defined as a const.